### PR TITLE
docs(chart):update KB for responsive Chart

### PR DESCRIPTION
**Pull request by NansiYancheva**
Related to: https://github.com/telerik/blazor/issues/9161

### CODE REVIEW

File: components/chart/overview.md [hunk 1]
```diff
@@ -94,7 +94,7 @@ You can add a descriptive text that enriches the [Title](#title) by adding the `
...
 
-You can also set the chart size in percentage values so it occupies its container when it renders. If the parent container size changes, you must call the chart's `Refresh()` C# method after the DOM has been redrawn and the new container dimensions are rendered. You can do this when you explicitly change container sizes (like in the example below), or from code that gets called by events like `window.resize`. You can find an example of making charts redraw on `window.resize` in the [Responsive Chart](https://github.com/telerik/blazor-ui/tree/master/chart/responsive-chart) sample.
+You can also set the chart size in percentage values so it occupies its container when it renders. If the parent container size changes, you must call the chart's `Refresh()` C# method after the DOM has been redrawn and the new container dimensions are rendered. You can do this when you explicitly change container sizes (like in the example below), or from code that gets called by events like `window.resize`. You can find an example and guideline of making Chart redraws on `window.resize` in the [knowledge base article for responsive Chart]({%slug chart-kb-responsive%}).
```
Comment (dimodi) [inline components/chart/overview.md:97]:
```suggestion
You can also set the chart size in percentage values so it occupies its container when it renders. If the parent container size changes, you must call the chart's `Refresh()` C# method after the DOM has been redrawn and the new container dimensions are rendered. You can do this when you explicitly change container sizes (like in the example below), or from code that gets called by events like `window.resize`. You can find an example and guidelines for making Charts refresh on `window.resize` in the knowledge base article for [responsive Chart]({%slug chart-kb-responsive%}).
```

File: knowledge-base/chart-responsive.md [hunk 2]
```diff
@@ -22,12 +22,28 @@ res_type: kb
...
 ## Solution
 
-Generally, the `Width` and `Height` parameters of the chart can take values in `%`, and the chart will render according to the dimensions of its parent element.
+Generally, the `Width` and `Height` parameters of the Chart can take values in `%`, and the Chart will render according to the dimensions of its parent element.
 
-This works well for the initial rendering and the chart will be "responsive" immediately according to your layout, regardless of the display (desktop, tablet, phone).
+This works well for the initial rendering and the Chart will be "responsive" immediately according to your layout, regardless of the display (desktop, tablet, phone).
 
-When the layout changes dynamically at runtime, you have to call its `.Refresh()` method. You can find an example in the following sample project: [https://github.com/telerik/blazor-ui/tree/master/chart/responsive-chart](https://github.com/telerik/blazor-ui/tree/master/chart/responsive-chart)
+When the layout changes dynamically at runtime, you have to call its `.Refresh()` method.
```
Comment (dimodi) [inline knowledge-base/chart-responsive.md:33]:
```suggestion
When the page layout changes dynamically at runtime, you have to call the [Chart `.Refresh()` method]({%slug components/chart/overview%}#chart-reference-and-methods). 
```

File: knowledge-base/chart-responsive.md [hunk 3]
```diff
@@ -22,12 +22,28 @@ res_type: kb
...
-Generally, the `Width` and `Height` parameters of the chart can take values in `%`, and the chart will render according to the dimensions of its parent element.
+Generally, the `Width` and `Height` parameters of the Chart can take values in `%`, and the Chart will render according to the dimensions of its parent element.
 
-This works well for the initial rendering and the chart will be "responsive" immediately according to your layout, regardless of the display (desktop, tablet, phone).
+This works well for the initial rendering and the Chart will be "responsive" immediately according to your layout, regardless of the display (desktop, tablet, phone).
 
-When the layout changes dynamically at runtime, you have to call its `.Refresh()` method. You can find an example in the following sample project: [https://github.com/telerik/blazor-ui/tree/master/chart/responsive-chart](https://github.com/telerik/blazor-ui/tree/master/chart/responsive-chart)
+When the layout changes dynamically at runtime, you have to call its `.Refresh()` method. 
+
+## Sample project
```
Comment (dimodi) [inline knowledge-base/chart-responsive.md:35]:
```suggestion
### Sample Project
```

File: knowledge-base/chart-responsive.md [hunk 4]
```diff
@@ -22,12 +22,28 @@ res_type: kb
...
+
+You can find an example in the following sample project: [https://github.com/telerik/blazor-ui/tree/master/chart/responsive-chart](https://github.com/telerik/blazor-ui/tree/master/chart/responsive-chart)
```
Comment (dimodi) [inline knowledge-base/chart-responsive.md:37]: Do not use URLs as link text. Not easy to read and not good for SEO.

```suggestion
You can find an example in the [Responsive Chart GitHub project](https://github.com/telerik/blazor-ui/tree/master/chart/responsive-chart)
```

File: knowledge-base/chart-responsive.md [hunk 5]
```diff
@@ -22,12 +22,28 @@ res_type: kb
...
+
+The project is a Blazor Web App developed on the .NET 8 framework. It is created by following the [steps in the Telerik Blazor Web App (Tutorial)]({%slug getting-started/web-app%}). It features server interactive render mode and a global interactivity location. The sample project offers an exemplary project structure. While individual projects may vary, it's crucial to ensure that correct naming conventions and references are used.
```
Comment (dimodi) [inline knowledge-base/chart-responsive.md:39]: Most of this paragraph is unnecessary and the end is hard to understand. If this is important, add it to the guidelines below.

File: knowledge-base/chart-responsive.md [hunk 6]
```diff
@@ -22,12 +22,28 @@ res_type: kb
...
-When the layout changes dynamically at runtime, you have to call its `.Refresh()` method. You can find an example in the following sample project: [https://github.com/telerik/blazor-ui/tree/master/chart/responsive-chart](https://github.com/telerik/blazor-ui/tree/master/chart/responsive-chart)
+When the layout changes dynamically at runtime, you have to call its `.Refresh()` method. 
+
+## Sample project
+
+You can find an example in the following sample project: [https://github.com/telerik/blazor-ui/tree/master/chart/responsive-chart](https://github.com/telerik/blazor-ui/tree/master/chart/responsive-chart)
+
+The project is a Blazor Web App developed on the .NET 8 framework. It is created by following the [steps in the Telerik Blazor Web App (Tutorial)]({%slug getting-started/web-app%}). It features server interactive render mode and a global interactivity location. The sample project offers an exemplary project structure. While individual projects may vary, it's crucial to ensure that correct naming conventions and references are used. 
+
+Sample project guideline:
```
Comment (dimodi) [inline knowledge-base/chart-responsive.md:41]:
```suggestion
### Implementation Guidelines
```

File: knowledge-base/chart-responsive.md [hunk 7]
```diff
@@ -22,12 +22,28 @@ res_type: kb
...
+The `RaiseWindowResizeEvent`method updates the window dimensions. The method also raises the `WindowResize` event, if subscribed to, to notify of the resize event.
+1. Add the `ResponsiveChartComponent.razor`. In the `OnInitialized` method subscribe to the `WindowResize` event and unsubscribe from the event in the `Dispose` method. This will ensure that the event is followed only when the component is in use. The `ResizeChart` method ensures that whenever the window resize event fires, the Chart refreshes and adapts to the new size of its container.
+1. For the sake of the example, the `ResponsiveChartComponent` is a model, which is in a separate razor file. In the `ResizeChart.razor` the `ResponsiveChartComponent` is used to create three individual responsive Charts that adjust dynamically based on the size of the browser window.
+1. The sample project also includes a responsive Chart print functionallity, which is handled only in the `PrintResponsiveChart.razor`. You can also check the [knowledge base article on the normal Chart print functionallity]({%slug chart-kb-print-chart-only%}).
```
Comment (dimodi) [inline knowledge-base/chart-responsive.md:49]: Avoid passive voice and use simpler phrases.

```suggestion
1. `PrintResponsiveChart.razor` also shows a responsive Chart print functionality. You can also check the knowledge base article about [printing Charts]({%slug chart-kb-print-chart-only%}).
```

File: knowledge-base/chart-responsive.md [hunk 8]
```diff
@@ -22,12 +22,28 @@ res_type: kb
...
+
+## Sample project
+
+You can find an example in the following sample project: [https://github.com/telerik/blazor-ui/tree/master/chart/responsive-chart](https://github.com/telerik/blazor-ui/tree/master/chart/responsive-chart)
+
+The project is a Blazor Web App developed on the .NET 8 framework. It is created by following the [steps in the Telerik Blazor Web App (Tutorial)]({%slug getting-started/web-app%}). It features server interactive render mode and a global interactivity location. The sample project offers an exemplary project structure. While individual projects may vary, it's crucial to ensure that correct naming conventions and references are used. 
+
+Sample project guideline:
+
+1. In the `wwwwroot` folder add the JS file `windowResizeHandler.js`. The JS file defines a function `raiseResizeEvent`, which listens for the [window resize event](https://developer.mozilla.org/en-US/docs/Web/API/Window/resize_event). When the resizing is detected the JS function invokes the `RaiseWindowResizeEvent` C# method. Ensure that the assembly name and method name in the function match the corresponding ones in your application.
```
Comment (dimodi) [inline knowledge-base/chart-responsive.md:43]: Most of the guidelines are too detailed and imply specific method names, types, life cycle methods, etc. Some of the guidelines are not necessary at all, for example the text about `~` - it's very unlikely someone to hit that problem. You can make the guidelines sound a bit more general and just point to what is done in which file.

File: knowledge-base/chart-responsive.md [hunk 9]
```diff
@@ -22,12 +22,24 @@ res_type: kb
 
 ## Description
 
-When the user resizes the browser window or some layout change happens (for example, a navigation panel is expanded or collapsed), you may want to have the charts redraw with the new dimensions.
+When the user resizes the browser window or some layout change happens (for example, a navigation panel is expanded or collapsed), you may want to have the Chart redraws with the new dimensions.
```
Comment (ikoevska) [inline knowledge-base/chart-responsive.md:25]:
```suggestion
When the user resizes the browser window or some layout change happens (for example, a navigation panel is expanded or collapsed), you may want to have the Chart redraw with the new dimensions.
```

File: knowledge-base/chart-responsive.md [hunk 10]
```diff
@@ -22,12 +22,24 @@ res_type: kb
...
 
-This works well for the initial rendering and the chart will be "responsive" immediately according to your layout, regardless of the display (desktop, tablet, phone).
+This works well for the initial rendering and the Chart will be "responsive" immediately according to your layout, regardless of the display (desktop, tablet, phone).
 
-When the layout changes dynamically at runtime, you have to call its `.Refresh()` method. You can find an example in the following sample project: [https://github.com/telerik/blazor-ui/tree/master/chart/responsive-chart](https://github.com/telerik/blazor-ui/tree/master/chart/responsive-chart)
+When the page layout changes dynamically at runtime, you have to call the [Chart `.Refresh()` method]({%slug components/chart/overview%}#chart-reference-and-methods). 
+
+### Sample Project
+
+You can find an example in the [Responsive Chart GitHub project](https://github.com/telerik/blazor-ui/tree/master/chart/responsive-chart)
```
Comment (ikoevska) [inline knowledge-base/chart-responsive.md:37]:
```suggestion
You can find an example in the [Responsive Chart GitHub project](https://github.com/telerik/blazor-ui/tree/master/chart/responsive-chart).
```

**Review by ikoevska — APPROVED** [review]

File: knowledge-base/chart-responsive.md [hunk 11]
```diff
@@ -22,12 +22,122 @@ res_type: kb
...
+Generally, the `Width` and `Height` parameters of the Chart can take values in `%`, and the Chart will render according to the dimensions of its parent element.
 
-This works well for the initial rendering and the chart will be "responsive" immediately according to your layout, regardless of the display (desktop, tablet, phone).
+This works well for the initial rendering and the Chart will be "responsive" immediately according to your layout, regardless of the display (desktop, tablet, phone).
 
-When the layout changes dynamically at runtime, you have to call its `.Refresh()` method. You can find an example in the following sample project: [https://github.com/telerik/blazor-ui/tree/master/chart/responsive-chart](https://github.com/telerik/blazor-ui/tree/master/chart/responsive-chart)
+When the window resizes, you have to resize the Chart dynamically at runtime:
+
+1. Add a JS function that listens for the [window resize event](https://developer.mozilla.org/en-US/docs/Web/API/Window/resize_event) and invokes a C# method. Ensure that the method name in the function match the one in your application.
```
Comment (ntacheva) [inline knowledge-base/chart-responsive.md:35]:
```suggestion
1. Add a JS function that listens for the [window resize event](https://developer.mozilla.org/en-US/docs/Web/API/Window/resize_event) and invokes a C# method. Ensure that the method name in the function matches the one in your application.
```
```suggestion
1. Add a JS function that listens for the [window resize event](https://developer.mozilla.org/en-US/docs/Web/API/Window/resize_event) and invokes a C# method. Ensure that the method name in the function match the one in your application.
```

File: knowledge-base/chart-print-chart-only.md [hunk 12]
```diff
@@ -23,9 +23,80 @@ res_type: kb
 
 ## Description
 
-How to print the rendered Blazor Chart?
+How to print the rendered Blazor Chart? I want to print single or separate Chart components.
```
Comment (ntacheva) [inline knowledge-base/chart-print-chart-only.md:26]: I don't quite understand the second sentence. Can we add some more details?

**Review by ntacheva — APPROVED** [review]

**Pull request merged**